src/patchkernel/surface_skd_tree.cpp

```diff
@@ -581,6 +497,220 @@ long SurfaceSkdTree::findPointClosestCell(int nPoints, const std::array<double,
+* the point and closest cellss. If all cells contained in the tree are
+* farther than the maximum distance, the argument will be set to the
+* maximum representable distance
+*/
+long SurfaceSkdTree::findPointClosestCells(const std::array<double, 3> &point, double maxDistance,
+                                          std::vector<long> &ids, double *distance) const
+{
+    return findPointClosestCells(point, maxDistance, false, ids, distance);
+}
+
+/*!
+* Given the specified point find the closest cells contained in the tree and
+* evaluate the distance between those cells and the given point. All the cells
+* at the same minimal distance (using patch tolerance) are returned.
+*
+* \param[in] point is the point
+* \param[in] maxDistance all cells whose distance is greater than
+* this parameters will not be considered for the evaluation of the
+* distance
+* \param[in] interiorCellsOnly if set to true, only interior cells will be considered,
+* it will be possible to consider non-interior cells only if the tree has been
+* instantiated with non-interior cells support enabled
+* \param[out] ids on output it will contain the ids of the closest cells.
+* If all cells contained in the tree are farther than the maximum
+* distance, the first element of the container will be set to the null id.
+* \param[out] distance on output it will contain the distance between
+* the point and closest cell. If all cells contained in the tree are
+* farther than the maximum distance, the argument will be set to the
+* maximum representable distance
+*/
+long SurfaceSkdTree::findPointClosestCells(const std::array<double, 3> &point, double maxDistance,
+                                          bool interiorCellsOnly, std::vector<long> &ids, double *distance) const
+{
+    // Tolerance for distance evaluations
+    const PatchKernel &patch = getPatch();
+    double tolerance = patch.getTol();
+
+    // Initialize the cell id
+    ids.resize(1, Cell::NULL_ID);
+
+    // Get the root of the tree
+    std::size_t rootId = 0;
+    const SkdNode &root = m_nodes[rootId];
+    if (root.isEmpty()) {
+        *distance = std::numeric_limits<double>::max();
+
+        return 0;
+    }
+
+    // If threads safe lookups are not needed, a temporary data structure
+    // are declared as member of the class to avoid its reallocation every
+    // time the function is called.
+    ClosestCellCandidates *closestCellCandidates = nullptr;
+    std::unique_ptr<ClosestCellCandidates> privateStorage = nullptr;
+    if (areLookupsThreadSafe()) {
+        privateStorage = std::unique_ptr<ClosestCellCandidates>(new ClosestCellCandidates());
+        closestCellCandidates = privateStorage.get();
+    } else {
+        closestCellCandidates = &m_closestCellCandidates;
+    }
+    closestCellCandidates->maxDistance = *distance;
+    findPointClosestCandidates(point, maxDistance, closestCellCandidates);
+
+    // Process the candidates and find the closest cells
```

Comment:
To make evaluation of the distance clearer, please move the initialization if the distance here:
```
*distance = closestCellCandidates->maxDistance;

long nDistanceEvaluations = 0;
for () {
  ... 
}
```
with the code inside the for cycle using distance instead of closestCellCandidates->maxDistance.